autopilot/financial-services/bankruptcy.ipynb

```diff
@@ -0,0 +1,728 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sklearn import metrics\n",
+    "\n",
+    "colors = [\"blue\", \"green\"]\n",
+    "model_names = [\"Objective : AUC\", \"Objective : F1\"]\n",
+    "models = [data_auc, data_f1]\n",
+    "for i in range(len(models)):\n",
+    "    fpr, tpr, _ = metrics.roc_curve(y_test, models[i][\"proba\"])\n",
+    "    fpr, tpr, _ = metrics.roc_curve(y_test, models[i][\"proba\"])\n",
+    "    auc_score = metrics.auc(fpr, tpr)\n",
+    "    plt.plot(\n",
+    "        fpr,\n",
+    "        tpr,\n",
+    "        label=str(f\"Auto Pilot {auc_score:.3f} {model_names[i]}\"),\n",
+    "        color=colors[i],\n",
+    "    )\n",
+    "\n",
+    "plt.xlim([-0.1, 1.1])\n",
+    "plt.ylim([-0.1, 1.1])\n",
+    "plt.ylabel(\"True Positive Rate\")\n",
+    "plt.xlabel(\"False Positive Rate\")\n",
+    "plt.legend(loc=\"lower right\")\n",
+    "plt.title(\"ROC Cuve\")"
+   ]
+  },
+  {
+   "cell_type": "raw",
+   "metadata": {},
+   "source": [
+    "The precision-recall curve compares the trade-off between precision and recall, with the best models having a precision-recall curve that is flat initially, dropping steeply as the recall approaches 1. The higher precision + recall, more the curve will be skewed towards upper right."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "colors = [\"blue\", \"green\"]\n",
+    "model_names = [\"Objective : AUC\", \"Objective : F1\"]\n",
+    "models = [data_auc, data_f1]\n",
+    "\n",
+    "print(\"model \", \"F1 \", \"precision \", \"recall \")\n",
+    "for i in range(0, len(models)):\n",
+    "    precision, recall, _ = precision_recall_curve(y_test, models[i][\"proba\"])\n",
+    "    print(\n",
+    "        model_names[i],\n",
+    "        f1_score(y_test, np.array(models[i][\"label\"])),\n",
+    "        precision_score(y_test, models[i][\"label\"]),\n",
+    "        recall_score(y_test, models[i][\"label\"]),\n",
+    "    )\n",
+    "    plt.plot(recall, precision, color=colors[i], label=model_names[i])\n",
+    "\n",
+    "plt.xlabel(\"Recall\")\n",
+    "plt.ylabel(\"Precision\")\n",
+    "plt.title(\"Precision-Recall Curve\")\n",
+    "plt.legend(loc=\"upper right\")\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "---\n",
+    "## Conclusion <a name=\"Conclusion\"></a>\n",
+    "We can see that with very little data science knowledge, we are able to create a moderately accurate prediction for a complex Financial event like Bankruptcy. From the AUC and Precision+Recall plots, we can also see that Autopilot handled highly imbalanced data resonably well. We think the reason for the 62% Recall (rather than achieving higher score) is as follows: the bankruptcy dataset is missing some important features of Bankruptcy filing - short term liquidity, short term funding source etc.    "
```

Comment:
```suggestion
    "You can see that with very little data science knowledge, you are able to create a moderately accurate prediction for a complex financial event like bankruptcy. From the AUC and Precision+Recall plots, you can also see that Autopilot handled highly imbalanced data reasonably well. We think the reason for the 62% Recall (rather than achieving higher score) is as follows: the bankruptcy dataset is missing some important features of bankruptcy filing - short term liquidity, short term funding source etc."
```